sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -446,6 +446,20 @@ static void appendCompileOptionsFromImage(std::string &CompileOpts,
       CompileOpts = NewCompileOpts;
       OptPos = CompileOpts.find(TargetRegisterAllocMode);
     }
+    static const char *FP32PrecDiv = "-foffload-fp32-prec-div";
+    if (auto Pos = CompileOpts.find(FP32PrecDiv); Pos != std::string::npos) {
+      const char *BackendOption = nullptr;
+      PlatformImpl->getBackendOption(FP32PrecDiv, &BackendOption);
+      auto OptLen = strlen(FP32PrecDiv);
+      CompileOpts.replace(Pos, OptLen, BackendOption);
+    }
+    static const char *FP32PrecSqrt = "-foffload-fp32-prec-sqrt";
+    if (auto Pos = CompileOpts.find(FP32PrecSqrt); Pos != std::string::npos) {
+      const char *BackendOption = nullptr;
+      PlatformImpl->getBackendOption(FP32PrecSqrt, &BackendOption);
+      auto OptLen = strlen(FP32PrecSqrt);
+      CompileOpts.replace(Pos, OptLen, BackendOption);
+    }
```

Comment:
Since they seem to be doing the same work, I suppose we could do something like:
```suggestion
    constexpr std::string_view ReplaceOpts[] = {"-foffload-fp32-prec-div", "-foffload-fp32-prec-sqrt"};
    for (const std::string_view Opt : ReplaceOpts) {
        if (auto Pos = CompileOpts.find(Opt); Pos != std::string::npos) {
            const char *BackendOption = nullptr;
            PlatformImpl->getBackendOption(Opt, &BackendOption);
            CompileOpts.replace(Pos, Opt.length(), BackendOption);
        }
    }
```
If we want to avoid the extra indentation, we could instead do:
```suggestion
    constexpr std::string_view ReplaceOpts[] = {"-foffload-fp32-prec-div", "-foffload-fp32-prec-sqrt"};
    for (const std::string_view Opt : ReplaceOpts) {
       auto Pos = CompileOpts.find(Opt);
       if ( Pos == std::string::npos)
         continue;
        const char *BackendOption = nullptr;
        PlatformImpl->getBackendOption(Opt, &BackendOption);
        CompileOpts.replace(Pos, Opt.length(), BackendOption);
      }
```
I don't have a preference, but I think consolidating the logic looks nice.

Reply:
applied with a slight change due to `const char *`ness of `getBackendOption` 1st argument, thanks!